src/dataServices/dynamoDbBundleService.ts

```
@@ -270,6 +286,81 @@ export class DynamoDbBundleService implements Bundle {
+                    return key.endsWith('.reference');
+                });
+                if (references.length === 0) {
+                    return;
+                }
+                const versionedLinksSet = new Set(versionedLinksArray);
+                await Promise.all(
+                    references.map(async (referencePath: string) => {
+                        if (!versionedLinksSet.has(referencePath)) {
+                            return;
+                        }
+                        let entryReference = flattenedResources[referencePath];
+                        const fullUrlMatch = entryReference.match(captureFullUrlParts);
+                        if (!fullUrlMatch) {
+                            return;
+                        }
+                        const resourceType = fullUrlMatch[2];
+                        const id = fullUrlMatch[3];
+                        let vid = fullUrlMatch[4];
+                        if (!vid) {
+                            const compoundId = `${resourceType}_${id}`;
+                            if (compoundId in idToVersionId) {
+                                vid = idToVersionId[compoundId];
+                            } else {
+                                const projectionExpression = 'meta';
+                                try {
+                                    const itemResponse: GenericResponse = await this.dynamoDbHelper.getMostRecentResource(
+                                        resourceType,
+                                        id,
+                                        projectionExpression,
+                                    );
+                                    const { meta } = itemResponse.resource;
+                                    vid = meta.versionId;
+                                } catch (e) {
+                                    logger.error('Failed to find most recent resource', e);
```

what should be do in this case? swallow the error and move on? or fail the transaction?

---

Good question. Nobody likes silent errors. It might hint at a programming error if those occur. It's probably safer to fail the tx.